includes/utilities.inc

```diff
@@ -158,12 +180,25 @@ function islandora_multi_importer_read_filedata($file_path, $numrows = 20, $offs
               $maxRow = $rowindex;
               break;
             }
-            $table[$rowindex] = $row;
+
+            if ($flat[0] != '#') {
+              $row[] = "$spreadsheetId : $range";   // Append the $spreadsheetId and $range to the end of the row as import_source. Addresses https://github.com/mnylc/islandora_multi_importer/issues/42
+              if ($append_import_index) { $row[] = $rowindex; }  // MAM. Append the $rowindex to the end of the row as import_index.  Addresses https://github.com/mnylc/islandora_multi_importer/issues/20
+              $table[$rowindex] = $row;  // Partially addresses https://github.com/mnylc/islandora_multi_importer/issues/15
+              $maxRow = $rowindex;
             }
-        $maxRow = $rowindex;
```

Comment:
sure you need to remove  $table[$rowindex] = $row; and $maxRow? here? this breaks some logic can you justify?